src/twitter/twitter-notifications/twitternotificationsyncadaptor.cpp

```diff
@@ -0,0 +1,628 @@
+        decrementSemaphore(accountId);
+        return;
+    }
+
+    bool ok = false;
+    QJsonArray tweets = parseJsonArrayReplyData(replyData, &ok);
+    if (ok) {
+        if (!tweets.size()) {
+            SOCIALD_LOG_DEBUG("no retweets received for account" << accountId);
+            decrementSemaphore(accountId);
+            return;
+        }
+
+        QString selfUserScreenName;
+        QHash<QString, int> dbRetweetCounts = m_db.retweetedTweetCounts(accountId);
+        QHash<QString, int> retweetCounts;
+        QStringList newlyRetweetedTweets;
+        int delta = 0;
+        foreach (const QJsonValue &tweetValue, tweets) {
+            QJsonObject tweet = tweetValue.toObject();
+            selfUserScreenName = tweet.value(QLatin1String("user")).toObject().value(QLatin1String("screen_name")).toString();
+            QString retweetId = tweet.value(QLatin1String("id_str")).toString();
+            int retweetsCount = tweet.value(QLatin1String("retweet_count")).toInt();
+            retweetCounts.insert(retweetId, retweetsCount);
+            if (!dbRetweetCounts.contains(retweetId) || dbRetweetCounts.value(retweetId) < retweetsCount) {
+                delta += retweetsCount - dbRetweetCounts.value(retweetId);
+                newlyRetweetedTweets.append(retweetId);
+            }
+        }
+
+        m_db.setRetweetedTweetCounts(accountId, retweetCounts); // won't get committed until finalize();
+        if (newlyRetweetedTweets.size() > 0) {
+            // Search if we already have a notification
+            Notification *notification = createNotification(accountId, Retweet);
+
+            // Set properties of the notification
+            QStringList openUrlArgs;
+            notification->setItemCount(newlyRetweetedTweets.size());
+            notification->setTimestamp(QDateTime::currentDateTimeUtc());
+            if (newlyRetweetedTweets.size() == 1) {
+                //: This label refers to a single Tweet by the user, eg: "Your tweet" + "has been retweeted 5 times"
+                //% "Your tweet"
+                notification->setSummary(qtTrId("qtn_social_notifications_twitter_your_tweet"));
+                //: This label tells the user how many times a single Tweet has been retweeted
+                //% "has been retweeted %n times"
+                notification->setBody(qtTrId("qtn_social_notifications_twitter_1_n_retweets", delta));
+                openUrlArgs << QLatin1String("https://mobile.twitter.com/") + selfUserScreenName + QLatin1String("/status/") + newlyRetweetedTweets.first();
+            } else {
+                //: This label refers to multiple Tweets by the user, eg: "Your tweets" + "have been retweeted 5 times"
+                //% "Your tweets"
+                notification->setSummary(qtTrId("qtn_social_notifications_twitter_your_tweets"));
+                //: This label tells the user how many times multiple Tweets have been retweeted
+                //% "have been retweeted %n times"
```

Comment:
right, this string was meant to be n_n_retweets instead of 1_n_retweets.  Will fix.